mlflow/pytorch/_lightning_autolog.py

```diff
@@ -560,17 +636,42 @@ def patched_predict(*args, **kwargs):
 
         client.flush(synchronous=False)
 
-        result = original(self, *args, **kwargs)
-
-        if early_stop_callback is not None:
-            _log_early_stop_metrics(early_stop_callback, client, run_id, model_id=model_id)
-
-        if Version(pl.__version__) < Version("1.4.0"):
-            summary = str(ModelSummary(self.model, mode="full"))
-        else:
-            summary = str(ModelSummary(self.model, max_depth=-1))
-
         with tempfile.TemporaryDirectory() as tempdir:
+            os.environ[_MLFLOW_LIGHTNING_AUTOLOGGING_TMP_DIR_ENV] = tempdir
+
+            try:
+                result = original(self, *args, **kwargs)
+            finally:
+                for callback in self.callbacks:
+                    if isinstance(callback, __MlflowPLCallback) and callback._model_forward_patch:
+                        gorilla.revert(callback._model_forward_patch)
+
+            model_signature = None
+            input_output_tensors_file = os.path.join(tempdir, "input_output_tensors.pkl")
+            if os.path.exists(input_output_tensors_file):
+                input_tensor, output_tensor = torch.load(input_output_tensors_file)
+                try:
+                    input_example = input_tensor.cpu().numpy()
+                    with torch.no_grad():
+                        output_example = output_tensor.cpu().numpy()
+                    model_signature = infer_signature(
+                        input_example,
+                        output_example,
+                    )
+                except Exception as e:
+                    _logger.warning(
+                        "Inferring model signature failed, skip logging signature. "
+                        f"root cause: {e!r}."
```

Comment:
Could we do a `from e` to preserve the ordering of the stack trace?

Reply:
`_logger.warning` can't use `from e`, but I added `exc_info=True` so that it can print error stack.